frontend/src/components/GrantAccess.vue

```diff
@@ -1,160 +1,13 @@
 <script setup lang="ts">
 import PageTitle from '@/components/PageTitle.vue';
-import { ApiService, type ApplicationRoleResponse, type GrantUserRoleRequest } from '@/services/ApiService';
-import { selectedApplication } from '@/services/ApplicationState';
-import { onMounted, ref } from 'vue';
-import { useToast } from 'vue-toastification';
-
-const FOREST_CLIENT_INPUT_MAX_LENGTH = 8
-const domainOptions = {IDIR: 'I', BCEID: 'B'} // TODO, load it from backend when backend has the endpoint.
-let applicationRoleOptions = ref<ApplicationRoleResponse[]>([])
-
-const defaultFormData = {
-  domain: domainOptions.BCEID,
-  userId: null,
-  forestClientNumber: null,
-  role: null as unknown as ApplicationRoleResponse
-}
-const formData = ref(JSON.parse(JSON.stringify(defaultFormData))) // clone default input data.
-
-const apiService = new ApiService()
-
-onMounted(async () => {
-  applicationRoleOptions.value = await apiService.getApplicationRoles(
-    selectedApplication?.value?.application_id
-  ) as ApplicationRoleResponse[]
-})
-
-function onlyDigit(evt: KeyboardEvent) {
-  if (isNaN(parseInt(evt.key))) {
-    evt.preventDefault()
-  }
-}
-
-async function grantAccess() {
-  const toast = useToast();
-  const grantAccessRequest = toRequest(formData.value)
-  try {
-    await apiService.grantUserRole(grantAccessRequest)
-    toast.success(`User "${grantAccessRequest.user_name}"" is granted with "${formData.value.role.role_name}" access.`)
-    formData.value = JSON.parse(JSON.stringify(defaultFormData)) // clone default input data.
-  }
-  catch(err: any) {
-    useToast().error(`Grant Access failed due to an error. Please try again.If the error persists then contact support.\nMessage: ${err.response.data?.detail}`)
-    console.error("err: ", err)
-  } 
-}
-
-function toRequest(formData: any) {
-  const request = {
-    user_name: formData.userId,
-    user_type_code: formData.domain,
-    role_id: formData.role.role_id,
-    ...(formData.forestClientNumber? 
-        {forest_client_number: formData.forestClientNumber.padStart(FOREST_CLIENT_INPUT_MAX_LENGTH, '0')}
-        : {})
-  } as GrantUserRoleRequest
-
-  return request
-}
-
+import GrantAccessForm from '../components/GrantAccessForm.vue';
```

Comment:
I'm not sure why you moved everything into a separate component?

Reply:
This is due to my testings, I am switching it back to the original component.